advisor/src/main/kotlin/advisors/VulnerableCode.kt

```diff
@@ -90,13 +90,14 @@ class VulnerableCode(name: String, vulnerableCodeConfiguration: VulnerableCodeCo
         val packageMap = packages.associateBy { it.purl }
         val packageVulnerabilities = service.getPackageVulnerabilities(PackagesWrapper(packageMap.keys))
 
-        return packageVulnerabilities.filter { it.unresolvedVulnerabilities.isNotEmpty() }.mapNotNull { pv ->
-            packageMap[pv.purl]?.let { pkg ->
-                val vulnerabilities = pv.unresolvedVulnerabilities.map { it.toModel() }
-                val summary = AdvisorSummary(startTime, Instant.now())
-                pkg to listOf(AdvisorResult(details, summary, vulnerabilities = vulnerabilities))
-            }
-        }.toMap()
+        return packageVulnerabilities.filter { it.unresolvedVulnerabilities.isNotEmpty() && it.purl.isNotEmpty() }
+            .mapNotNull { pv ->
+                packageMap[pv.purl]?.let { pkg ->
+                    val vulnerabilities = pv.unresolvedVulnerabilities.map { it.toModel() }
+                    val summary = AdvisorSummary(startTime, Instant.now())
+                    pkg to listOf(AdvisorResult(details, summary, vulnerabilities = vulnerabilities))
+                }
+            }.toMap()
```

Comment:
Nit: I guess our preferred reformatting would be like

        return packageVulnerabilities.filter {
            it.unresolvedVulnerabilities.isNotEmpty() && it.purl.isNotEmpty()
        }.mapNotNull { pv ->
            packageMap[pv.purl]?.let { pkg ->
                val vulnerabilities = pv.unresolvedVulnerabilities.map { it.toModel() }
                val summary = AdvisorSummary(startTime, Instant.now())
                pkg to listOf(AdvisorResult(details, summary, vulnerabilities = vulnerabilities))
            }
        }.toMap()

as it reduces indentation changes and thus keeps the diff smaller.

Reply:
The filtering has to be done in a different place, therefore this change is no longer in place.